tools/openmm/pdbfixer.xml

```diff
@@ -0,0 +1,189 @@
+            $replace_nonstandard                 ## replace nonstandard residues with standard equivalents
+            $add_residues                        ## add missing residues
+            --ph='$ph'                           ## the pH to use for adding missing hydrogens [default: 7.0]
+
+            #if $solvent.box.water_box == "true":
+                --water-box=$solvent.box.x $solvent.box.y $solvent.box.z ## add a water box. The value is the box dimensions in nm [example: --water-box=2.5 2.4 3.0]
+                --positive-ion='$solvent.positive_ion' ##   positive ion to include in the water box: Cs+, K+, Li+, Na+, or Rb+ [default: Na+]
+                --negative-ion='$solvent.negative_ion' ## negative ion to include in the water box: Cl-, Br-, F-, or I- [default: Cl-]
+                --ionic-strength='$solvent.ionic_strength' ## molar concentration of ions to add to the water box [default: 0.0]
+            #end if
+            --verbose
+
+    ]]></command>
+    <inputs>
+        <param format="pdb" name="pdb_input" type="data" label="PDB input file"/>
+        <param name="add_atoms" type="select" label="Missing atoms to be added">
+            <option value="all" selected="true">All</option>
+            <option value="heavy">Heavy atoms only</option>
+            <option value="hydrogen">Hydrogen atoms only</option>
+            <option value="none">None</option>
+        </param>
+        <param name="keep_heterogens" type="select" label="Which heterogens to keep">
+            <option value="all" selected="true">All</option>
+            <option value="water">Only water</option>
+            <option value="none">None</option>
+        </param>
+        <param name="replace_nonstandard" type="boolean" label="Replace nonstandard residues with standard equivalents?" truevalue="--replace-nonstandard" falsevalue="" value=""/>
+        <param name="add_residues" type="boolean" label="Add missing residues?" truevalue="--add-residues" falsevalue="" value=""/>
+        <param name="ph" type="float" min="0" max="14" value="7" label="pH" help="Ignored if not adding hydrogen atoms"/>
+        <section name="solvent" title="Solvent parameters" help="Note that you may prefer to modify solvent parameters using other tools.">
+            <conditional name="box">
+                <param name="water_box" type="select" label="Add a water box?" help="Dimensions in nanometers">
+                    <option value="true">Yes</option>
+                    <option value="false" selected="true">No</option>
+                </param>
+                <when value="true">
+                    <param name="x" type="float" min="0" value="0" label="Size in X dimension"/>
+                    <param name="y" type="float" min="0" value="0" label="Size in Y dimension"/>
+                    <param name="z" type="float" min="0" value="0" label="Size in Z dimension"/>
+                </when>
```

Comment:
I'm not familiar with this program, but is a rectangular box the only option?

Reply:
Yes, it seems so. If users want something else they should use editconf. I'll make a note of that in the help.